### ISSUE TITLE
Overall structure and style

### ISSUE THREAD

**Issue by igarizio** [signalling-games-org/evoke#75]
Hi, I have basically checked every piece of code and everything looks pretty good. I especially appreciate all the comments and academic references everywhere! Here are some minor suggestions regarding code structure and overall "polished-ness":

- Ideally, your `main` branch should not contain code that is under development, TODO comments, etc. [Here are some TODO examples](https://github.com/search?q=repo%3Asignalling-games-org%2Fevoke%20TODO&type=code) (obviously not all of them are relevant) and [here is another one](https://github.com/signalling-games-org/evoke/blob/9b94aa6bfb1e5a9286b35734377c6d9a1c5ea5f5/evoke/src/evolve.py#L712) that also seems important. You can obviously have code under development in other branches.
 - [Commented-out/unused imports ](https://github.com/signalling-games-org/evoke/blob/main/evoke/src/figure.py#L25-L28)
 - [Unfinished docstrings](https://github.com/signalling-games-org/evoke/blob/main/evoke/src/evolve.py#L163) ([here is another example](https://github.com/signalling-games-org/evoke/blob/9b94aa6bfb1e5a9286b35734377c6d9a1c5ea5f5/evoke/src/evolve.py#L436)) [Small disclaimer here: It seems like you are following numpydoc, I usually don't follow that one so there is a chance these cases are ok and I just don't know it]
 - [Error/Warning logging through `print` statements](https://github.com/signalling-games-org/evoke/blob/9b94aa6bfb1e5a9286b35734377c6d9a1c5ea5f5/evoke/src/games.py#L275)
 - [Random spacings](https://github.com/signalling-games-org/evoke/blob/9b94aa6bfb1e5a9286b35734377c6d9a1c5ea5f5/evoke/src/games.py#L320)

As I said in the beginning, the library seems **pretty solid** and these are (very) minor suggestions, but small things like these might discourage users from trusting your code (and we don't want that!).

**Comment by stephenfmann** [comment]
Thanks, working on all these now.

**Comment by stephenfmann** [comment]
Thank you for your extremely in-depth comments! I have worked through the repository and responded to each point below:

- **TODO comments and main branch**: I've removed almost all TODOs and shifted the remaining ones to a separate branch, which I'll work through after release. There should be no more TODOs left in the main branch at all; I'm waiting for GitHub to update since the last commit (the search seems to be cached somehow) but this should be done now.
- **Unused imports**: I have removed all commented-out imports.
- **Unfinished docstrings**: I have gone through all the source and example scripts and I believe every function now has a complete docstring, including parameters and return values. 
- **Errors and print() statements**: I'm going to push this until after the next release. I've created a separate issue for it.
- **Random spacings**: I think I've caught all of these, and I have run the entire repository through the `black` formatter for good measure.

I hope this is all good! Thanks again and let me know if you have any further suggestions. If you're happy I'll close the issue.
